tests/test_params.py

```diff
@@ -53,6 +53,7 @@ def test_hyperparameterset_normalize_runs():
     normalized_runs = valid_set.normalize_runs_as_array([r1, r2])
     assert normalized_runs.shape == (2, 1)
 
+    # TODO(gst): fix test, or allow for this case. When can this happen?
     valid_set = HyperParameterSet(
         [
             HyperParameter("v1", {"value": 1}),
```

Comment:
What does the test below try to achieve? I'm not quite sure when this could happen. It breaks under the new stricter flow. We were logging this as a warning before but letting 0 be the value.

Reply:
@dannygoldstein  Thoughts on this?

Reply:
I don't think the sweep can generate run configs with subsets of the hyperparameters, else that would be conditional logic etc, so I think we should get rid of this test...

Reply:
nice catch. it looks like the previous version of the test was checking whether a run without a value of all parameters specified in the sweep config can be properly converted to a vector. the only path to getting sweep runs that are missing parameters defined in the sweep config that I can think of is by using the "prior runs" feature of sweeps to create a new sweep and including runs that do not have parameters defined in the sweep config. i just tried this and while the sweep indeed runs, it never terminates at least for a grid search. 

i think the right thing to do here is to allow the sweep to handle runs missing config parameters (the prior runs case) but to also ensure to properly terminate the sweep in such cases. here is a repro:

https://wandb.ai/dannygoldstein/uncategorized/sweeps/2fsfdazr/overview

I created this sweep using those prior runs which are missing some of the sweep params. 

Running that with a training script that just prints `wandb.config` you can see i get way more runs than i should (4), and the prior runs that do correctly contain the parameters are ignored

Reply:
ignore the part above about stopping, my example uses a bayes sweep and therefore shouldn't stop